sdk/eidolon_ai_sdk/builtins/logic_units/vectara.py

```diff
@@ -0,0 +1,83 @@
+import os
+from urllib.parse import urljoin
+
+from httpx import AsyncClient
+from pydantic import BaseModel, Field
+
+from eidolon_ai_sdk.apu.logic_unit import LogicUnit, llm_function
+from eidolon_ai_sdk.system.reference_model import Specable
+
+
+class VectaraSearchSpec(BaseModel):
+    """
+    A logic unit for searching in Vectara. Requires the VECTARA_API_KEY environment variable to be set for authentication.
+    """
+
+    corpus_key: str = Field(description="The corpus key to search in.")
+    description: str = Field("Search documents related to {corpus_key}.", description="Description of the tool presented to LLM. Will be formatted with corpus_key.")
+    vectara_url: str = "https://api.vectara.io/"
```

Comment:
Vectara tool for agentic rag. Allows agents to 1. query arbitrary search terms and 2. read full documents as desired (doc id is retrieved from searches)